src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+
+    def forward_train(
+        self,
+        feature_maps,
+        img_metas,
+        proposal_list,
+        gt_bboxes,
+        gt_labels,
+        gt_bboxes_ignore=None,
+        gt_masks=None,
+    ):
+        """
+        Args:
+            feature_maps (`List[torch.Tensor]`):
+                List of multi-level image features.
+            img_metas (`List[Dict]`):
+                List of image info dict where each dict. Has: 'img_shape', 'scale_factor', 'flip', and may also contain
+                'filename', 'ori_shape', 'pad_shape', and 'img_norm_cfg'. For details on the values of these keys see
+                `mmdet/datasets/pipelines/formatting.py:Collect`.
+            proposals (`List[torch.Tensor]`):
+                List of region proposals.
+            gt_bboxes (`List[torch.Tensor]`):
+                Ground truth bboxes for each image with shape (num_ground_truths, 4) in [top_left_x, top_left_y,
+                bottom_right_x, bottom_right_y] format.
+            gt_labels (`List[Tensor]`):
+                Class indices corresponding to each box.
+            gt_bboxes_ignore (`List[torch.Tensor]`, *optional*):
+                Specify which bounding boxes can be ignored when computing the loss.
+            gt_masks (`List[torch.Tensor]`, *optional*) :
+                True segmentation masks for each box used if the architecture supports a segmentation task.
+
+        Returns:
+            dict[str, Tensor]: a dictionary of loss components
+        """
+        # assign gts and sample proposals
+        if self.with_bbox or self.with_mask:
+            num_imgs = len(img_metas)
+            if gt_bboxes_ignore is None:
+                gt_bboxes_ignore = [None for _ in range(num_imgs)]
+            sampling_results = []
+            for i in range(num_imgs):
+                assign_result = self.bbox_assigner.assign(
+                    proposal_list[i], gt_bboxes[i], gt_bboxes_ignore[i], gt_labels[i]
+                )
+                sampling_result = self.bbox_sampler.sample(
+                    assign_result,
+                    proposal_list[i],
+                    gt_bboxes[i],
+                    gt_labels[i],
+                    feats=[lvl_feat[i][None] for lvl_feat in feature_maps],
+                )
+                sampling_results.append(sampling_result)
+
+        losses = {}
+        # bbox head forward and loss
+        if self.with_bbox:
+            bbox_results = self._bbox_forward_train(feature_maps, sampling_results, gt_bboxes, gt_labels)
+
+            losses.update(bbox_results["loss_bbox"])
+
+        if self.with_mask:
+            mask_results = self._mask_forward_train(
+                feature_maps, sampling_results, bbox_results["bbox_feats"], gt_masks
+            )
+            losses.update(mask_results["loss_mask"])
+
+        return losses
+
+    def forward_test(self, hidden_states, proposal_list):
+        """Test without augmentation (originally called `simple_test`).
```

Comment:
We should be careful with this as:

* We're not applying augmentation at any point, so this description could be confusing
* This method is used when there aren't any labels passed, but wouldn't be called if the model is in eval mode and labels were passed